src/nix-env/nix-env.cc

```diff
@@ -224,6 +224,100 @@ static void checkSelectorUse(DrvNames & selectors)
 }
 
 
+namespace {
+
+template <class Container, class F>
+void forEachWithIndex(Container & container, F && func) {
+    std::size_t index = 0;
+    for (auto & item : container) {
+        func(item, index);
+        ++index;
+    }
+}
+
+std::set<std::string> fuzzySearch(const DrvInfos & allElems, const DrvName & name) {
```

Comment:
Maybe "fuzzy" is not a great name since it's just prefix matching. But if we had a fuzzy search (like using Levenshtein distance) it would be great to move this function into `libutil` so it could be used for other things like attribute/variable name mismatches.

Reply:
True. Renamed to `searchByPrefix` to better reflect the current situation. Real fuzzy search would be nicer but I guess it could be left to be improved at a later stage

Reply:
Factor out, then improve, is always good! :)